modin/engines/base/frame/partition_manager.py

```diff
@@ -249,6 +249,8 @@ def broadcast_axis_partitions(
         # partitions as best we can right now.
         if keep_partitioning:
             num_splits = len(left) if axis == 0 else len(left.T)
+        elif lengths:
```

Comment:
If we process `lengths` here, do we need to pass it into `apply` method below as `kw`?

Reply:
It is not necessary, because in the case when `right_partitions != None`, then we get into a function that cannot work with lengths - `deploy_func_between_two_axis_partitions`.